### ISSUE TITLE
macOS dynamic framework unit tests fail (crash) due to missing token, although a token is supplied

### ISSUE THREAD

**Issue by akitchen** [mapbox/mapbox-gl-native#10641]
When running the tests of the dynamic framework on macOS, I seem to always run into this error, which results in a crash of the test suite:
```
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: You must provide a Mapbox API access token for Mapbox tile sources
```
I've tried putting my token in the test bundle's Info.plist as well as in the target's environment. Neither seems to be working. Also, it seems to be a side-effect of a test which has already finished executing, not the test name displayed at the end of the log (which is often different)

**Comment by 1ec5** [comment]
Oddly, I’ve only reproduced this issue once in Xcode 9.2.

**Comment by akitchen** [comment]
Seems like it's probably related to #10768

**Comment by 1ec5** [comment]
I got this in the console, which might be a clue:

```
2017-12-20 23:32:02.624021-0800 xctest[21217:11455112] [ERROR] {com.mapbox.mbgl.Map Snapshotter}[Setup]: loading style failed: HTTP status code 401
```

**Comment by akitchen** [comment]
Fixed on `release-agua` branch via #10776
